src/widgets/chart/lv_chart.c

```diff
@@ -462,8 +462,19 @@ void lv_chart_set_next_value(lv_obj_t * obj, lv_chart_series_t * ser, int32_t va
 
     lv_chart_t * chart  = (lv_chart_t *)obj;
     ser->y_points[ser->start_point] = value;
+    // If optimization isn't enabled or chart data series aren't updated synchronously, this part makes disturbances.
+    if(chart->type == LV_CHART_TYPE_LINE && chart->update_mode == LV_CHART_UPDATE_MODE_CIRCULAR) {
+        for(int32_t i = 1; i < 10; i++) {
```

Comment:
@kisvegabor Well, you know that I am a newbie. And this change was just a suggestion. You are more familiar with this library. The problem I had encountered is that on the implementation on hardware, if one wants to use this feature it is really hard. Moreover, this chart doesn't provide smooth drawing in crowded mode but only vertical lines. so in case of smooth signals like sinus, it isn't really plotted as smooth as sin (especially when the space or area is not much.). more description is that I had tried different methods, for example to add some points then spaces and then redraw chart (in different functions). This resulted in that if the number of data added is less than the space, then the graph drawing would become piecewise rather being continuous.

Reply:
This is beyond my knowledge of this library. I will try to dig in more in this library. I think I should start with the implementation of the core. But in this special case, I think we can use the idea behind the qcustomplot of QT or the qtcharts, since they have the smoothing option, and I don't know if they use the bezier or spline method or any other method. I don't know if there is any other open-source chart or graph library available.

Reply:
Sorry, I missed your comment :slightly_frowning_face: 

We don't need to deal with easing now. 

I think what we should in this PR:
- Keep the changes in `draw_series_line`
- Add a function for deleting `N` values in `lv_example_chart_8()` (this function can be a reference for the users)